### PR TITLE
json-merge-patch, add test case for map's key is `null`

### PR DESCRIPTION
**Pull request by haolingdong-msft**
When a map's key is `null`, it should throw error when doing serialization both in `toJson()` and `toJsonMergePatch()`.
We will throw `NullPointorException` when map's key value is `null`. This PR add a test case for this.

### CODE REVIEW

File: typespec-tests/src/test/java/com/cadl/patch/PatchClientTest.java [hunk 1]
```diff
@@ -73,6 +73,21 @@ public void testSerializationForMapProperty() throws JsonProcessingException {
...
+            Resource resource = new Resource(new HashMap<>());
+            JsonMergePatchHelper.getResourceAccessor().prepareModelForJsonMergePatch(resource, true);
+            resource.getMap().put(null, new InnerModel("value1"));
+            BinaryData.fromObject(resource).toString();
+        });
+
+        String expectedMessage = "'fieldName' cannot be null.";
+        String actualMessage = exception.getMessage();
+
+        Assertions.assertTrue(actualMessage.contains(expectedMessage));
```
Comment (weidongxu-microsoft) [inline typespec-tests/src/test/java/com/cadl/patch/PatchClientTest.java:88]: Not sure if we want to verify the message. `fieldName` a bit strange to me?

**Review by weidongxu-microsoft — APPROVED** [review]

File: typespec-tests/src/test/java/com/cadl/patch/PatchClientTest.java (hunk 1 above)
Comment (haolingdong-msft) [reply typespec-tests/src/test/java/com/cadl/patch/PatchClientTest.java:88]: I add this because I want to verify it's the same error message with non-json-merge-patch cases. But I do agree the message is a bit strange.

Comment (weidongxu-microsoft) [reply typespec-tests/src/test/java/com/cadl/patch/PatchClientTest.java:88]: Fine for me. It should be from core, hope they won't change that.
